### ISSUE TITLE
complain when IFakeOptionsBuilder.Implements is passed a non-interface

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#462]
It's supposed to allow us to add interfaces to a fake, but it accepts types.

```c#
A.Fake<IInterface>(builder => builder.Implements(typeof (SomeClass)));
```

passes silently, but with no obvious benefit.

Casting to `SomeClass` fails.

Found while investigating an answer to [Having an interface fake inherit from abstract while both share same interface inheritance](http://stackoverflow.com/questions/28697918/fakeiteasy-having-an-interface-fake-inherit-from-abstract-while-both-share-sam).

**Comment by blairconrad** [comment]
Changing to **enhancement** since FakeItEasy is doing nothing wrong when used as intended. This is just a bonus warning for users to help them use it correctly.

**Comment by jimmyheaddon** [comment]
Anyone mind if I take this?  Seems like a relatively simple modification for my first contribution

**Comment by blairconrad** [comment]
Hi, @jimmyheaddon.
Not at all. Thanks for the interest. I'll take a gander at the pull request as soon as I can.

**Comment by blairconrad** [comment]
Fixed by #536.

**Comment by blairconrad** [comment]
@jimmyheaddon, we often thank people on Twitter and add them to a "contributors" list.
Based on your GitHub handle and avatar you're using here (which I'm taking as a hint to your geographical location), I think I know who you are there.
Am I right?
Do you mind?

In related news, thanks for your hard work.
Look for your name in the release notes! :trophy:

**Comment by jimmyheaddon** [comment]
@blairconrad feel free, I think I'm the only person of my namesake on Twitter and LinkedIn :smile:

Hopefully the next PR won't be such hard work for you guys, now I have a better understanding how you work!

**Comment by adamralph** [comment]
@jimmyheaddon on the contrary, it's always a pleasure to help get a new contributor on board.

**Comment by blairconrad** [comment]
Ditto. It's been a delight, @jimmyheaddon.

**Comment by blairconrad** [comment]
@jimmyheaddon, thanks very much for your work on this issue. Look for your name in the release notes. :trophy:

This issue has been fixed in [release 2.0.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.0.0).
